src/shared/smack-util.c

```diff
@@ -76,10 +77,14 @@ int mac_smack_apply(const char *path, SmackAttr attr, const char *label) {
         if (!mac_smack_use())
                 return 0;
 
+        fd = openat(dir_fd, path, O_PATH|O_CLOEXEC|O_NOFOLLOW);
+        if (fd < 0)
+                return -errno;
+
         if (label)
-                r = lsetxattr(path, smack_attr_to_string(attr), label, strlen(label), 0);
+                r = fsetxattr(fd, smack_attr_to_string(attr), label, strlen(label), 0);
```

Comment:
this doesn't (or didn't?) work on O_PATH.

redirect to mac_smack_apply_fd() instead, which goes via `FORMAT_PROC_FD_PATH(fd)`